packages/acs-ui-common/src/identifier.ts

```diff
@@ -37,7 +53,23 @@ export const toFlatCommunicationIdentifier = (id: CommunicationIdentifier): stri
  * @public
  */
 export const fromFlatCommunicationIdentifier = (id: string): CommunicationIdentifier => {
-  // This implementation is currently a hack that only works with ACS identifiers.
-  // TODO: Make `toFlatCommunicationIdentifier` not be lossy so this can reverse the process.
-  return { communicationUserId: id };
+  if (id.startsWith(COMMUNICATION_USER_PREFIX)) {
+    return { communicationUserId: id.substr(COMMUNICATION_USER_PREFIX.length) };
```

Comment:
How come we previously didn't need to strip out the prefix? (e.g. we call `call?.removeParticipant(fromFlatCommunicationIdentifier(userId))` which will now have a different id being used)

Reply:
`toFlatCommunicationIdentifer` wasn't adding a prefix. Because it was doing a lossy stringification -- just the ID in all cases without any prefix.